### DISCUSSION TITLE
Route width

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#52] [Q&A] [answered]
Can it be that the auto zoom scale in percent in the navigation menu has no effect? I have changed the display size and now would like to extend the route width from 220% to 300%. But it has no effect at all.

**Comment by devemux86** [comment]
You mention 3 different settings?
- Settings | Application | Advanced | Display size
- Settings | Routing | Appearance | Route width
- Settings | Navigation | Auto zoom | Auto zoom scale

The question is about the "Auto zoom scale"? It works with "Auto zoom mode | Speed".
Have you changed enough the min + max auto zoom levels or they limit the auto zoom?

**Comment by devemux86** [comment]
For example:
- Setting "Auto zoom scale" **<** 100% may need to set smaller "Minimum auto zoom level"
- Setting "Auto zoom scale" **>** 100% may need to set larger "Maximum auto zoom level"

So there is plenty of zoom space for the auto zoom to work.

---

There can be improvements in _speed_ auto zoom, so I plan to review the implementation.

**Comment by DolomitesTom** [comment, marked as answer]
Settings | Routing | Appearance | Route width

was what I was searching for, thanks a lot.

Need to play a little bit with the other settings to understand them.

At the moment display size 150%
Settings | Navigation | Auto zoom | Auto zoom scale
220% Min 16.6 max 18.6 and depending on the speed. In this situation I didn't recognize a difference of auto zoom scale up to 300%

**Reply by devemux86** [reply]
> Settings | Navigation | Auto zoom | Auto zoom scale 220% Min 16.6 max 18.6 and depending on the speed. In this situation I didn't recognize a difference of auto zoom scale up to 300%

I recommend to unlock (for testing) the min / max auto zoom levels: set min 12 and max 20.
Then you can make some tests to see how the speed auto zoom works with your speed.

So you can see the full zoom range inside it chooses to work and we can look for improvements.

**Reply by DolomitesTom** [reply]
I can give it a try. But I guess we had this topic already in the past. I would need it the opposite way: big zoom when driving fast and low zoom when driving slow (what hardly ever happens🙄)

**Comment by MotoUKRider** [comment]
I favour landscape for navigation, using auto-zoom (speed) at 15.0 to 18.0.
Settings | Application | Display | Advanced | Display size = 120% (for the bigger side-bar and larger on-map buttons)

I make use of "Follow" mode when not navigating, and it seems to be just right with these settings.  I think trial+error is needed here to determine how much you want to see when riding a long/fast road (ie: no turn due) and navigating a residential area with many side-roads.

**Comment by DolomitesTom** [comment]
Have now set min zoom to 12, max to 20, zoom depending on speed, off-route ,  zoom scale 200%.

max 20 is in principle my preferred setting, therefore is it possible to enlarge the max setting to 30? Default setting can stay. 
While navigation it was always far away from 12. I would say all zooming happened between 19 and 20, even at speed of 140 km/h. So I don't understand the behaviour. Why should I set minimum lower than 18 if this is never used?

**Reply by devemux86** [reply]
> is it possible to enlarge the max setting to 30?

It cannot be larger than the max zoom of the map: offline maps 20, online maps 18.

Offline maps are vector maps, so zoom is unlimited, practically we set a sane max.
Zoom level 20 is around map scale 10m.

> Why should I set minimum lower than 18 if this is never used?

I suggested setting min / max at their ends to see _without limits_ where the auto zoom works.
Then you can limit it according to your preferences.

---

Speed auto zoom may need improvements, I plan to review it in the future.
Currently, the auto zoom scale option moves the zoom range up or down.

Maybe we need a different implementation where we set the min / max zoom
and speed auto zoom moves between them, covering the whole zoom range.

**Reply by DolomitesTom** [reply]
Thanks a lot for reviewing the speed auto zoom.

**Comment by WolfgangZ42** [comment]
Despite a review is not a bad thing - I'm happy with AutoZoom as it is currently. 

Using display size 130, autozoom min/max 12.0/20.0 and scale 150 I get a zoom level from ~15 to ~19
Changing to scale 50 results in a zoom level from ~13 to ~18

Here a test with the same route, same position and approximately the same speed:

<img src="https://user-images.githubusercontent.com/121806238/221373740-fcf60d06-8c23-41f1-b3b5-c4fb13757c9b.jpg" width="200"/>  <img src="https://user-images.githubusercontent.com/121806238/221373750-2a01691f-e40b-4ab3-be70-c3a4635924b4.jpg" width="200"/>  <img src="https://user-images.githubusercontent.com/121806238/221373761-5dabb37d-d356-45a1-9f01-4c9e35cfd334.jpg" width="200"/>
This is my favored setting since speed related autozoom was announced (scale 150)


<img src="https://user-images.githubusercontent.com/121806238/221373768-0b7fcd38-df00-465c-ac49-8bbbee00b4da.jpg" width="200"/>  <img src="https://user-images.githubusercontent.com/121806238/221373784-eff2dd01-fa48-4ed1-a0b9-4d88cf33314c.jpg" width="200"/>  <img src="https://user-images.githubusercontent.com/121806238/221373792-3525547a-4958-419f-b119-d10f92a349c1.jpg" width="200"/>
And this the test with scale 50 - (I forgot to hide the buttons 😄 )

**Comment by devemux86** [comment]
About the speed auto zoom please see https://github.com/devemux86/cruiser/discussions/63.
